include/bredis/impl/connection.ipp

```diff
@@ -154,6 +154,9 @@ Connection<NextLayer>::read(DynamicBuffer &rx_buff,
     namespace sys = boost::system;
 
     auto rx_bytes = asio::read_until(stream_, rx_buff, MatchResult(1), ec);
+    if (ec) {
+        return positive_parse_result_t{{}, 0};
+    }
 
     const char *char_ptr =
         boost::asio::buffer_cast<const char *>(rx_buff.data());
```

Comment:
Actually, this won't work. The return value of `data()` is a **ConstBufferSequence** not a *ConstBuffer*. It could be a list of buffers. It just happens to work for you because `boost::asio::basic_streambuf` returns a `boost::asio::const_buffers_1` from `data()`. This code will not compile if `rx_buff` is a `beast::basic_dynabuf` for example. It seems like you're doing conversion to `std::string` more often than necessary (maybe). Does `Protocol::parse` require a linear buffer? You have some choices:

1. Rewrite `Protocol::parse` to take the input as a **ConstBufferSequence** template type. You can use `boost::asio::buffers_iterator` to adapt the buffer sequence to a sequential access iterator range.

2. Detect when `data()` returns multiple buffers, and do a dynamic allocation to flatten the buffer into one contiguous block of memory. This is the approach I used in Beast:
https://github.com/vinniefalco/Beast/blob/2904bf57fe345549b9ea1b54400bfbec65b124c2/include/beast/http/impl/basic_parser.ipp#L261